### PR TITLE
add mentee route and status validation, unit test were previously in …

### LINKED ISSUE #31: Implement Get all mentees by status endpoint (Admin)
**Description:**
This issue involves implementing an API to get all mentees by status. The endpoint should allow Admin to make a GET request to `{{baseUrl}}/api/admin/mentees?status=ONGOING`.

Example get all approved mentees
```
{
  "mentees": [
    {
      "mentee_id": 1,
      "created_at": "2023-06-30T08:00:00Z",
      "updated_at": "2023-06-30T12:30:00Z",
      "state": "APPROVED",
      "answers": {
        "question1": "answer1",
        "question2": "answer2"
      },
      "profile": {
        "created_at": "2023-06-29T09:45:00Z",
        "updated_at": "2023-06-30T14:15:00Z",
        "primary_email": "mentee@example.com",
        "contact_email": "mentee_contact@example.com",
        "first_name": "Sarah",
        "last_name": "Johnson",
        "image_url": "https://example.com/mentee_profile_image.jpg",
        "linkedin_url": "https://www.linkedin.com/in/sarahjohnson",
        "type": "DEFAULT",
        "uuid": "98765432-2345-6789-2345-678923456789"
      },
      "mentor": {
        "mentor_id": 1,
        "created_at": "2023-06-30T10:00:00Z",
        "updated_at": "2023-06-30T14:30:00Z",
        "state": "APPROVED",
        "category": "Technology",
        "application": {
          "field1": "value1",
          "field2": "value2"
        },
        "availability": true
      },
      "certificate_id": 123,
      "blogs": {
        "blog1": "https://example.com/blog1",
        "blog2": "https://example.com/blog2"
      }
    }
  ]
}
```
Example get all rejected mentees
```
{
  "mentees": [
    {
      "mentee_id": 2,
      "created_at": "2023-06-25T12:00:00Z",
      "updated_at": "2023-06-29T10:30:00Z",
      "state": "REJECTED",
      "answers": {
        "question1": "answer3",
        "question2": "answer4"
      },
      "profile": {
        "created_at": "2023-06-24T09:15:00Z",
        "updated_at": "2023-06-29T13:45:00Z",
        "primary_email": "mentee2@example.com",
        "contact_email": "mentee2_contact@example.com",
        "first_name": "Michael",
        "last_name": "Smith",
        "image_url": "https://example.com/mentee2_profile_image.jpg",
        "linkedin_url": "https://www.linkedin.com/in/michaelsmith",
        "type": "DEFAULT",
        "uuid": "67891234-3456-7890-3456-789034567890"
      },
      "mentor": null,
      "certificate_id": null,
      "blogs": ""
    }
  ]
}
```

Example get all pending mentees
```
{
  "mentees": [
    {
      "mentee_id": 2,
      "created_at": "2023-06-25T12:00:00Z",
      "updated_at": "2023-06-29T10:30:00Z",
      "state": "PENDING",
      "answers": {
        "question1": "answer3",
        "question2": "answer4"
      },
      "profile": {
        "created_at": "2023-06-24T09:15:00Z",
        "updated_at": "2023-06-29T13:45:00Z",
        "primary_email": "mentee2@example.com",
        "contact_email": "mentee2_contact@example.com",
        "first_name": "Michael",
        "last_name": "Smith",
        "image_url": "https://example.com/mentee2_profile_image.jpg",
        "linkedin_url": "https://www.linkedin.com/in/michaelsmith",
        "type": "DEFAULT",
        "uuid": "67891234-3456-7890-3456-789034567890"
      },
      "mentor": null,
      "certificate_id": null,
      "blogs": {}
    }
  ]
}
```

**Tasks:**
1. Create a controller for the `/controllers/mentee` endpoint in the backend (create the route mentee if not created).
2. Parse and validate the request body to ensure it matches the expected format.
3. Implement appropriate error handling and response status codes for different scenarios (e.g., validation errors, database errors).
5. Write unit tests to validate the functionality and correctness of the endpoint.

API documentation: https://documenter.getpostman.com/view/27421496/2s93m1a4ac#8744a3ee-970f-489a-853d-8b23fdee8de3

ER diagram: https://drive.google.com/file/d/11KMgdNu2mSAm0Ner8UsSPQpZJS8QNqYc/view 

**Acceptance Criteria:**
- The profile API endpoint `/controllers/mentee` is implemented via a GET request.
- The request body is properly parsed and validated for the required format.
- Appropriate error handling is implemented, providing meaningful error messages and correct response status codes.
- Unit tests are written to validate the functionality and correctness of the endpoint.

**Additional Information:**
No

**Related Dependencies or References:**
No

### PR DESCRIPTION
**Pull request by dsmabulage**
…the code

## Purpose

The purpose of this PR is to fix #31

## Goals

1. Add route mapping
2. Add TODO comment for invalid route path
3. Add status ENUM validation for mentee status
4. Unit tests were already there.

## Approach

1. Add small route mapping for mentees
2. Method was already written in the existing code
3. Add status validation for query param

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/routes/admin/mentee/mentee.route.ts [hunk 1]
```diff
@@ -9,8 +9,10 @@ import {
 
 const menteeRouter = express.Router()
 
-menteeRouter.get('/emails/', requireAuth, getAllMenteeEmails)
-menteeRouter.get('/applications', requireAuth, getMentees)
+menteeRouter.get('/', requireAuth, getMentees)
+menteeRouter.get('/emails', requireAuth, getAllMenteeEmails)
+// TODO: Remove this route
+// menteeRouter.get('/applications', requireAuth, getMentees)
```
Comment (anjula-sack) [inline src/routes/admin/mentee/mentee.route.ts:15]: can you remove this and send a PR to the frontend?
```suggestion
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/routes/admin/mentee/mentee.route.ts (hunk 1 above)
Comment (dsmabulage) [reply src/routes/admin/mentee/mentee.route.ts]: updated

**Comment by anjula-sack** [comment]
Impressive work @dsmabulage, Would you like to join our engineering team? You can read more about us here: https://handbook.sefglobal.org/engineering-team/team

**Comment by dileepainivossl** [comment]
> Impressive work @dsmabulage, Would you like to join our engineering team? You can read more about us here: https://handbook.sefglobal.org/engineering-team/team

Thank you for the opportunity! I'm very interested and will check out the link. Looking forward to learning more about the team!

**Comment by dsmabulage** [comment]
> Impressive work @dsmabulage, Would you like to join our engineering team? You can read more about us here: https://handbook.sefglobal.org/engineering-team/team

Thank you for the opportunity! I'm very interested and will check out the link. Looking forward to learning more about the team!

**Pull request closed without merging**